### ISSUE TITLE
Hard limits to time & distance traveled between points of subsequent frames

### ISSUE THREAD

**Issue by StefanMPopp** [mooch443/trex#26]
**Problem**
1) Tracks jumping around over long distances & times, even though I have set a max speed (which is the max observed speed for my animal).
2) Animals enter & exit from the same hole in my setup, so an animal disappearing into the hole and another reappearing thereafter are connected

**Possible solutions?**
1) an option for a hard cm limit of how far a track can jump.
2) an option for making tracks always terminate after track_max_reassign_time has passed

Is something like this possible?

**Comment by mooch443** [comment]
Hey, thanks for your report.

You might be able to circumvent your problem by setting `track_speed_decay = 1` and `track_max_reassign_time` to a high value (in seconds). This should essentially model the behaviour that you expected - identities might still jump to nearby objects, but in general should (given a low `track_max_speed`) not jump around too much. `track_speed_decay` provides a hint to TRex of what is expected to happen if an individual is lost - e.g. it will stay "in-place", or it will continue moving (over or behind obstacles) and emerge somewhere else. A value of 1 will indicate that it stays where it was last seen.

Does this help? 
-Tristan

**Comment by StefanMPopp** [comment]
Yes, this does help, thanks!
I'm not sure how best to communicate with you and I don't want to open lots of requests, so I'm going to add some offtopic here:
1) Is it possible to give the output columns a different name? I already managed to only output the columns I need.
2) The `track_ignore` format description is not clear to me: what does the `(, ...)` mean? Is there a `]` too many? I tried `[[1,1],[500,1],[500,500],[1,500]]` (as an example), but it threw me a syntax error.

**Comment by mooch443** [comment]
Hey,

here is good. Opening issues is (luckily) not an issue. It is actually good to have this as a reference for future readers as well. Parts could also be added to a Wiki in the future... we will see.

> 1. Is it possible to give the output columns a different name? I already managed to only output the columns I need.

If you are exporting as CSV, then renaming the columns would be simply to edit the (text-format) CSV files and change the name. This could also be batched of course (e.g. with Bash or Python). If you are using the standard way, you can load files in Python via (not tested):

```
import numpy as np
map = { 'old_column': 'new_column' }
file = {}
with np.load("fish0.npz") as npz:
    for key in map:
        file[map[key]] = npz[key]
np.savez("renamed_fish0.npz", *map)
```

> 2. The `track_ignore` format description is not clear to me: what does the `(, ...)` mean? Is there a `]` too many? I tried `[[1,1],[500,1],[500,500],[1,500]]` (as an example), but it threw me a syntax error.

In TRex (and better in the new version v1.1.2) you can CTRL+Click or CMD+Click or SHIFT+CLICK (depending on your system and TRex version, sorry) on the background to select a shape. This shape can then be either printed to the terminal (and you can copy it there) or directly added to `track_ignore` or `track_include`. You may have to reanalyse with the new shapes in place - but whatever shapes you select/print out should be compatible with those parameters.

Otherwise, or more generally, the syntax is:
```
[SHAPE,SHAPE,SHAPE, ... ]
```
and each SHAPE is:
```
[POINT,POINT, ... ]
```
So, in short, I believe you are missing a bracket pair around the list.

**Comment by StefanMPopp** [comment]
Thank you, that answers both questions! I had a rename script like this one in place, but wondered if it could be skipped.
I suggest you add a `[` in the examples for `track_ignore` and `track_include` and an 'each' after the '>= 3 points' in the descriptions (or something similar) to make it clearer.
